src/main/webapp/scripts/toggle.js

```diff
@@ -9,14 +9,17 @@ const BOOKS_OBJ_URL =  '/books';
 const PROJECTS_OBJ_URL = '/actions/projects';
 const OBJECTS_URLS = [BOOKS_OBJ_URL, PROJECTS_OBJ_URL];
 
+let loadingCounter;
```

Comment:
What's the purpose of this variable other than just getting updated with a new value?

Reply:
The purpose is to have a global loading decorator for the regular counter. Every time an object finishes loading and it's rendered it calls loadingCounter.decrement(). The decorator allows to update the progress bar and hide the loading cursor when completed.